### ISSUE TITLE
Docker crashes Windows on reboot after install. Laptop overheats!

### ISSUE THREAD

**Issue by FredUK** [docker/for-win#13505]
tldr; After installing Docker and rebooting, Windows is stuck on black screen. Laptop overheats and fans start spinning at max speed.
Seems bery similar to issue from 2018-2020 where many people were having similar experience [#1549 crashed my windows :((](https://github.com/docker/for-win/issues/1549)

After experiencing the same issue a few months back I have tried it again today, now running the latest Windows 10 22H2 and installed Docker 4.19. After reboot, it would be stuck on a black screen, after a few seconds I could hear the laptop fans spinning faster and faster. Putting my hand over the keyboard felt really hot. I can only assume the laptop was overheating but there was no way to see what it was doing.

Several restarts would not help, I had to go into the bios and **disable** CPU virtualization. Only then Windows booted fine. Quickly open HWMonitor and CPU temp was at 95deg (c) .

I'm not sure what it is but I can't get Docker Desktop installed on this machine.

  - [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
After reboot, machine does not boot, stuck on a black screen, high cpu usage, overheats and fans at max power.

### Expected behavior
Windows would boot fine after installing Docker and rebottingWindows.

### Information

  - Windows Version: Windows 10 22H2 (Windows 10 Home and later upgraded to Windows 10 Enterprise) = same issue
  - Docker Desktop Version: 4.19 and earlier
  - WSL2 or Hyper-V backend?  Both
  - Asus TUF Laptop (Ryzen 5 Mobile  3550H)
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
I have disabled CPU virtualization in order to boot
```
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[SKIP] DD0028: is there available VM disk space?
[SKIP] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[FAIL] DD0001: is the application running? Docker is not running
[SKIP] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[SKIP] DD0021: is the WSL 2 Windows Feature enabled?
[FAIL] DD0024: is WSL installed? exit status 0xfffffe44
[FAIL] DD0025: are WSL distros installed? WSL Distribution docker-desktop is missing
[PASS] DD0026: is the WSL LxssManager service running?
[FAIL] DD0029: is the WSL 2 Linux filesystem corrupt? exit status 0xffffffff
[FAIL] DD0035: is the VM time synchronized? exit status 0xffffffff
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[FAIL] DD0011: are the LinuxKit services running? failed to ping VM diagnosticsd with error: Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.
[2023-05-24T16:18:58.553738700Z][com.docker.diagnose.exe][I] ipc.NewClient: f655f630-diagnose -> \\.\pipe\dockerDiagnosticd diagnosticsd
[2023-05-24T16:18:58.554254900Z][com.docker.diagnose.exe][I] (4a516c63) f655f630-diagnose C->S diagnosticsd GET /ping
[2023-05-24T16:18:58.554769500Z][com.docker.diagnose.exe][W] (4a516c63) f655f630-diagnose C<-S NoResponse GET /ping (514.6µs): Get "http://ipc/ping": open \\.\pipe\dockerDiagnosticd: The system cannot find the file specified.

[FAIL] DD0004: is the Docker engine running? Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:18:58.557350900Z][com.docker.diagnose.exe][I] ipc.NewClient: b715530e-com.docker.diagnose -> \\.\pipe\dockerLifecycleServer VMDockerdAPI
[2023-05-24T16:18:58.558394000Z][com.docker.diagnose.exe][I] (46cda1b0) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /docker
[2023-05-24T16:18:58.558909200Z][com.docker.diagnose.exe][W] (46cda1b0) b715530e-com.docker.diagnose C<-S NoResponse GET /docker (515.2µs): Get "http://ipc/docker": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:18:58.559422800Z][com.docker.diagnose.exe][I] (46cda1b0-1) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:18:58.559935900Z][com.docker.diagnose.exe][W] (46cda1b0-1) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (513.1µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:18:59.564555200Z][com.docker.diagnose.exe][I] (46cda1b0-2) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:18:59.565075100Z][com.docker.diagnose.exe][W] (46cda1b0-2) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (519.9µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:00.571529500Z][com.docker.diagnose.exe][I] (46cda1b0-3) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:00.572083000Z][com.docker.diagnose.exe][W] (46cda1b0-3) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (553.5µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:01.577465500Z][com.docker.diagnose.exe][I] (46cda1b0-4) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:01.578024100Z][com.docker.diagnose.exe][W] (46cda1b0-4) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (558.6µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:02.579271900Z][com.docker.diagnose.exe][I] (46cda1b0-5) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:02.579873100Z][com.docker.diagnose.exe][W] (46cda1b0-5) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (601.2µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:03.591083300Z][com.docker.diagnose.exe][I] (46cda1b0-6) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:03.591686500Z][com.docker.diagnose.exe][W] (46cda1b0-6) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (603.2µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:04.603679000Z][com.docker.diagnose.exe][I] (46cda1b0-7) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:04.604290500Z][com.docker.diagnose.exe][W] (46cda1b0-7) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (611.5µs): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.
[2023-05-24T16:19:05.611014400Z][com.docker.diagnose.exe][I] (46cda1b0-8) b715530e-com.docker.diagnose C->S VMDockerdAPI GET /ping
[2023-05-24T16:19:05.612621800Z][com.docker.diagnose.exe][W] (46cda1b0-8) b715530e-com.docker.diagnose C<-S NoResponse GET /ping (1.6074ms): Get "http://ipc/ping": open \\.\pipe\dockerLifecycleServer: The system cannot find the file specified.

[PASS] DD0015: are the binary symlinks installed?
[FAIL] DD0031: does the Docker API work? error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine_linux/v1.24/containers/json": open //./pipe/docker_engine_linux: The system cannot find the file specified.
[PASS] DD0013: is the $PATH ok?
error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.24/containers/json": open //./pipe/docker_engine: The system cannot find the file specified.
[FAIL] DD0003: is the Docker CLI working? exit status 1
[PASS] DD0005: is the user in the docker-users group?
[FAIL] DD0038: is the connection to Docker working? HTTP GET https://login.docker.com: Get "https://login.docker.com": proxyconnect tcp: open \\.\pipe\dockerHTTPProxy: The system cannot find the file specified.
2023/05/24 17:19:07 error checking if default wsl distro is running: exit status 0xffffffff
```

### Steps to reproduce the behavior
1. Install docker
2. Restart

**Comment by samuk190** [comment]
wait until the memory leak be fixed, it's been a issue for more than 3 years, without that fix, reporting anything related to crash or overheating is useless because the memory leak could be the reason behind it

**Comment by goesbyabhi** [comment]
Ok this issue is pretty similar to mine and I fixed it yesterday. Actually this is not the issue of Docker but it's a hardware issue. And for me, whenever I went to the `Windows Optional Features` and turned on `Virtual Machine Platform` and `Windows Hypervisor Platform` then after reboot my laptop's screen would go black.

The actual reason why this is happening is because of UMA buffer size in the BIOS. The minimum buffer size for Hypervisor is 1G (1Gigs) but my laptop had set the buffer size to 512M (512Mb). You do not want to give it anything less than 1G. If you can't find 1G in the options then just set it  to auto. I did that and it worked fine for me. And I have now successfully installed WSL2 and Docker both.

**Comment by FredUK** [comment]
> Ok this issue is pretty similar to mine and I fixed it yesterday. Actually this is not the issue of Docker but it's a hardware issue. And for me, whenever I went to the `Windows Optional Features` and turned on `Virtual Machine Platform` and `Windows Hypervisor Platform` then after reboot my laptop's screen would go black.
> 
> The actual reason why this is happening is because of UMA buffer size in the BIOS. The minimum buffer size for Hypervisor is 1G (1Gigs) but my laptop had set the buffer size to 512M (512Mb). You do not want to give it anything less than 1G. If you can't find 1G in the options then just set it to auto. I did that and it worked fine for me. And I have now successfully installed WSL2 and Docker both.

You are very much correct! I had tried multiple things (including downgrading bios) - because I remember it used to work with CPU Virtualization before. When I switched from UMA Buffer size from 512mb to Auto, the machine booted correctly. Hats off to you ! 🥇

**Comment by jtromans** [comment]
I am running Windows 10 Pro 10.0.19045 Build 19045 with an Intel i9-9900X on an ASUS Sage x299 motherboard.

I installed Docker Desktop via Docker Desktop Installer with default settings, that include 'Use WSL 2 instead of Hyper-V (recommended)'. After an apparent successful installation and restart, I ran the Docker Desktop application only to find my entire system froze. Eventually I'm presented with DPC_Watchdog_violation and eventually my machine reboots.

Eventually I deduced that I should disable Intel(VMX) Virtualization Technology via my Asus UEFI Bios and this appears to bring stability to my machine. 

Worryingly, if I leave VMX enabled, my machine now permanently freezes, even in BIOS and before loading into Windows!

What on earth has Docker done to me? Could this be related to this issue?

**Comment by FredUK** [comment]
> What on earth has Docker done to me? Could this be related to this issue?

Sorry for the late reply. Just noticed this now. Yep, same thing happened to me. If I waited long enough when the machine froze the fan would start spinning at max speed and the laptop overheat. I'm no longer using Docker in Windows so I'm no longer worried. xD

**Comment by hieu22pham** [comment]
> > What on earth has Docker done to me? Could this be related to this issue?
> 
> Sorry for the late reply. Just noticed this now. Yep, same thing happened to me. If I waited long enough when the machine froze the fan would start spinning at max speed and the laptop overheat. I'm no longer using Docker in Windows so I'm no longer worried. xD



> > Ok this issue is pretty similar to mine and I fixed it yesterday. Actually this is not the issue of Docker but it's a hardware issue. And for me, whenever I went to the `Windows Optional Features` and turned on `Virtual Machine Platform` and `Windows Hypervisor Platform` then after reboot my laptop's screen would go black.
> > The actual reason why this is happening is because of UMA buffer size in the BIOS. The minimum buffer size for Hypervisor is 1G (1Gigs) but my laptop had set the buffer size to 512M (512Mb). You do not want to give it anything less than 1G. If you can't find 1G in the options then just set it to auto. I did that and it worked fine for me. And I have now successfully installed WSL2 and Docker both.
> 
> You are very much correct! I had tried multiple things (including downgrading bios) - because I remember it used to work with CPU Virtualization before. When I switched from UMA Buffer size from 512mb to Auto, the machine booted correctly. Hats off to you ! 🥇

You said that your machine worked correctly, why you didn't use Docker in Windows anymore? Does this way really fix the problem?

**Comment by samuk190** [comment]
> > > What on earth has Docker done to me? Could this be related to this issue?
> > 
> > 
> > Sorry for the late reply. Just noticed this now. Yep, same thing happened to me. If I waited long enough when the machine froze the fan would start spinning at max speed and the laptop overheat. I'm no longer using Docker in Windows so I'm no longer worried. xD
> 
> > > Ok this issue is pretty similar to mine and I fixed it yesterday. Actually this is not the issue of Docker but it's a hardware issue. And for me, whenever I went to the `Windows Optional Features` and turned on `Virtual Machine Platform` and `Windows Hypervisor Platform` then after reboot my laptop's screen would go black.
> > > The actual reason why this is happening is because of UMA buffer size in the BIOS. The minimum buffer size for Hypervisor is 1G (1Gigs) but my laptop had set the buffer size to 512M (512Mb). You do not want to give it anything less than 1G. If you can't find 1G in the options then just set it to auto. I did that and it worked fine for me. And I have now successfully installed WSL2 and Docker both.
> > 
> > 
> > You are very much correct! I had tried multiple things (including downgrading bios) - because I remember it used to work with CPU Virtualization before. When I switched from UMA Buffer size from 512mb to Auto, the machine booted correctly. Hats off to you ! 🥇
> 
> You said that your machine worked correctly, why you didn't use Docker in Windows anymore? Does this way really fix the problem?

Hi if you use experimental feature of WSL
%userprofile%
.wslconfig
[wsl2]
memory=10GB
processors=16
swap=100GB
localhostForwarding=true
nestedVirtualization=true

[experimental]
autoMemoryReclaim=dropcache

if you add [experimental] like mine, you will not suffer that issue anymore, been using this for months had 0 rpoblems since
Also I'm using Docker inside WSL, not docker desktop

**Comment by brian316** [comment]
docker desktop crashed and would not reboot. I installed the latest update to see if it would help and nothing. restarted pc and now windows 11 broke. I get a black screen and en error about twinapi.dll error.

**Comment by dsanchezchu** [comment]
Hi after install docker desktop, restarted pc, is automatically intalled  windows update and crash my laptop  is asus gl65, rtx 2060, i7 10750H attach image this error code .

any solution?
![Image](https://github.com/user-attachments/assets/421c6d76-d6ce-447e-ab6b-7706947086b6)

**Comment by brian316** [comment]
I had to reinstall windows 😄

**Comment by dsanchezchu** [comment]
When I load the system it shows that 80% of my CPU is being used, so I uninstalled it and it worked XD.

**Comment by amsadeeq** [comment]
![IMG_20250916_180948_812.jpg](https://github.com/user-attachments/assets/9a0abbcf-0ddd-4bb0-a056-a110847063dd)

Please I am installing docker unfortunately my windows is taking longer than expected to reboot. Kindly help please
